### DISCUSSION TITLE
Use of SPI for Ethernet / WiFi

### DISCUSSION THREAD

**Discussion by ghost** [grblHAL/STM32F4xx#89] [Ideas]
Hi Team,

I have recently been doing some design work around ESP32 and noticed some Ethernet and WiFi modules that can communicate over the SPI interface.
https://www.wiznet.io/product-item/wiz850io/
https://www.wiznet.io/product-item/wizfi360/

Is it possible to create a plugin (or even have this embedded as part of the standard firmware) that will allow the use of these types of devices which would then add greater flexibility to mCU's that do not normally have these options?

Cheers
G

**Comment by terjeio** [comment]
Adding a plugin is possible provided that a driver level interface is defined. E.g. the [EEPROM plugin](https://github.com/grblHAL/Plugin_EEPROM) relies upon such an interface for I2C comms.

Data structures/function prototypes used by driver level interfaces should be added to [plugins.h](https://github.com/grblHAL/core/blob/master/plugins.h).

**Comment by ghost** [comment]
OK. Is this something that can be done easily? I am not a coder and have no idea on how to do this.

I am hoping that someone from this community would do this as an enhancement.

Cheers and have a great day
